src/Datadog.Trace.ClrProfiler.Managed/Integrations/AspNetIntegration.cs

```diff
@@ -0,0 +1,48 @@
+#if !NETSTANDARD2_0
+
+using System;
+using System.Web;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.ClrProfiler.Integrations
+{
+    /// <summary>
+    ///     AspNetIntegration used to inject AspNetHttpModule IHttpModule into the application pipeline on startup
+    /// </summary>
+    public static class AspNetIntegration
+    {
+        private static readonly AspNetHttpModule AspNetHttpModule = new AspNetHttpModule("aspnet.request");
+        private static readonly ILog Log = LogProvider.GetLogger(typeof(AspNetIntegration));
+
+        /// <summary>
+        ///     Calls the underlying Init() For an HttpApplication and traces the request.
+        /// </summary>
+        /// <param name="thisObj">The HttpApplication instance ref.</param>
+        [InterceptMethod(
+            TargetAssembly = "System.Web",
+            TargetType = "System.Web.HttpApplication")]
+        public static void Init(object thisObj)
+        {
+            try
+            {
+                var initMethodAction = DynamicMethodBuilder<Action<object>>.GetOrCreateMethodCallDelegate(thisObj.GetType(), "Init");
```

Comment:
I used the IL as I can only actually call the Init() directly if the cast is successful.  Maybe we don't care if the cast isn't successful I suppose (i.e. just let it throw an exception), but that didn't seem right to me.  That said, I can easily obviously move the IL emt into that portion of the if block also

Reply:
I'm not so sure about the ordering, as the only example I can find orders it the same way, even though it does seem odd (i.e.init the application first, module 2nd).

Reply:
Whenever possible, we should let the C# compiler generate the IL for us instead of writing it by hand at run time and calling it through a delegate (but avoid using `dynamic` because that's actually worse). 

C# compiler > emit IL > `dynamic` > plain old reflection

Also, if the cast to `HttpApplication` ever fails, we have bigger problems to worry about 😉